### PR TITLE
Jump higher on trampoline when groundpounding

### PR DESCRIPTION
**Pull request by mrkubax10**
This pull request adds feature from #1724

### CODE REVIEW

File: src/object/trampoline.cpp [hunk 1]
```diff
@@ -85,13 +85,15 @@ Trampoline::collision(GameObject& other, const CollisionHit& hit)
     auto player = dynamic_cast<Player*> (&other);
     //Trampoline works for player
     if (player) {
+      if (player->m_does_buttjump)
+        player->m_does_buttjump = false;
```
Comment (Semphriss) [inline src/object/trampoline.cpp:89]: TODO: Check the pros and cons of this

Comment (mrkubax10) [reply src/object/trampoline.cpp:89]: This is workaround for this: [video](https://cdn.discordapp.com/attachments/396022213100437504/859345418441850910/2021-06-29_10-09-15.mp4). `m_does_buttjump` is set back to true later anyway.

Comment (Semphriss) [reply src/object/trampoline.cpp:89]: Ah, okay. In this case I retract my comment.

**Comment by Guih48** [comment]
This is a good idea, but it's makes many levels too easy, so i would suggest a new trampoline (or just bump block?) type that if you ground-pound it launches you high.

**Comment by mrkubax10** [comment]
Actually jump height is slightly smaller than normal jump. Try doing this in latest version or nightly and check what is happening.

**Comment by Semphriss** [comment]
Seems to work as intended, merging.

**Pull request merged**